Added /pizza page

DCoxshall opened the pull request:
Added pizza page, which currently displays some recent data handed to me over discord. Currently features two graphs, a bar chart and a pie chart. 
This PR is an attempt to solve [this](https://github.com/CSSUoB/cssuob.github.io/issues/329) issue. Unfortunately, I couldn't find a way link data from any easily-updatable source (i.e. Google Sheets) as was mentioned in the original issue. If this is something that needs to be done, then it's out of my ability range, for now.
Thanks for letting me take a stab at this. I know it's not much, but it's my first PR on a production codebase.

MattyTheHacker commented:
The light text on the dark background is a little hard to read
![image](https://user-images.githubusercontent.com/18513864/193162091-bf6041be-8e62-40cf-b34b-95e354400b2a.png)

other than that it looks pretty good to me. 

I'd also note that when hovering over the pie chart it only shows the number, not what the category is so might be nice to have a key saying which colour is which and/or changing it so hovering over it shows which flavour it is

LMBishop commented on pizzas.html:
```diff
@@ -0,0 +1,142 @@
+
+<!-- Annual bar chart scripting -->
+<script>
+  const chartBackgroundColors = [
+                'rgba(255, 99, 132, 0.2)',
+                'rgba(54, 162, 235, 0.2)',
+                'rgba(255, 206, 86, 0.2)',
+                'rgba(75, 192, 192, 0.2)',
+                'rgba(153, 102, 255, 0.2)',
+                'rgba(255, 159, 64, 0.2)',
+                'rgba(23, 43, 250, 0.2)'
+            ];
```
The colours look a bit muted compared to the rest of CSS' colours, though it's not really a massive issue

LMBishop commented on pizzas.html:
```diff
@@ -0,0 +1,142 @@
+    {% endfor %}
+   
+    <div id="annualBarChartHolder">
+      <canvas id="annualBarChart"></canvas>
```
It is worth adding `aria-label` and `role` attributes here (or something similar) for accessibility. See: https://www.chartjs.org/docs/latest/general/accessibility.html

For example `role="img" aria-label="Vegan Vegi Supreme: X, etc."`, or just saying it's a bar chart instead.

LMBishop reviewed (commented):
Looks good :)

Just a couple of comments below, and one other thing: could it be renamed to `pizza`? Better yet, have `pizzas` redirect to `pizza` too

DCoxshall commented:
> The light text on the dark background is a little hard to read ![image](https://user-images.githubusercontent.com/18513864/193162091-bf6041be-8e62-40cf-b34b-95e354400b2a.png)
> 
> other than that it looks pretty good to me.
> 
> I'd also note that when hovering over the pie chart it only shows the number, not what the category is so might be nice to have a key saying which colour is which and/or changing it so hovering over it shows which flavour it is

DCoxshall commented:
wait i closed it by accident

DCoxshall commented:
Ok so I fixed everything that people mentioned:
- I've turned the labels on the chart axes white
- I've added ARIA text to both charts
- I've made the colours bolder
- I've renamed the new page to /pizza

jedevc commented:
:eyes: I don't think we need the `package.json` and `package-lock.json` files committed? The page seems to pull by CDN (which is the way to go for this imo).

`node_modules` should definitely *never* be committed.

DCoxshall commented:
Okay I'm currently learning what "rebasing" is, I will fix this, just give me a minute.

jedevc commented on css/pizzas.css:
```diff
@@ -0,0 +1,15 @@
+#pizzaCharts {
+	width: 100%;
+	height: 100%;
+	display: flex;
```
I think we probably want to have a `@media` query somewhere here in this file, so that on mobile devices we can change the `flex-direction` to `column`, so that the pie chart shows below the bar chart on mobile, instead of disappearing off the edge of the screen.

DCoxshall commented:
Hey if this is good to go could someone review it please? If there's anything that needs changing lmk.

MattyTheHacker commented:
The pie chart still goes off the edge of the screen rather than moving below

DCoxshall commented:
Ok, how's this? I think I've finally got it fixed.

MattyTheHacker commented:
Looks good! there's a conflict with the redirects file since we changed it since you forked but if you can resolve that conflict I think this is god

MattyTheHacker commented on _data/redirects.yaml:
```diff
@@ -103,6 +77,7 @@
   destination: https://www.youtube.com/watch?v=WK5tCf7m-ZM
 - source: /soon
   destination: /calendar
+<<<<<<< HEAD
```
You've left in the conflict markers here and at the end of the file

JTWWilson replied:
If everything else is good you can just cut that out when merging

MattyTheHacker reviewed (approved):
okay yep this looks good, I'll wait for another review before merging just to be sure but yeah looks good

JTWWilson reviewed (approved):
Looks good, greta job @DCoxshall!
